contrib/storage-mongo/src/main/java/org/apache/drill/exec/store/mongo/plan/RexToMongoTranslator.java

```diff
@@ -0,0 +1,242 @@
+    .put(SqlStdOperatorTable.SIN, "$sin")
+    .put(SqlStdOperatorTable.SQRT, "$sqrt")
+    .put(SqlStdOperatorTable.SUBSTRING, "$substr")
+    .put(SqlStdOperatorTable.PLUS, "$add")
+    .put(SqlStdOperatorTable.MINUS, "$subtract")
+    .put(SqlStdOperatorTable.TAN, "$tan")
+    .put(SqlStdOperatorTable.TRIM, "trim")
+    .put(SqlStdOperatorTable.TRUNCATE, "$trunc")
+    .put(SqlStdOperatorTable.AND, MongoOp.AND.getCompareOp())
+    .put(SqlStdOperatorTable.OR, MongoOp.OR.getCompareOp())
+    .put(SqlStdOperatorTable.NOT, MongoOp.NOT.getCompareOp())
+    .put(SqlStdOperatorTable.EQUALS, MongoOp.EQUAL.getCompareOp())
+    .put(SqlStdOperatorTable.NOT_EQUALS, MongoOp.NOT_EQUAL.getCompareOp())
+    .put(SqlStdOperatorTable.GREATER_THAN, MongoOp.GREATER.getCompareOp())
+    .put(SqlStdOperatorTable.GREATER_THAN_OR_EQUAL, MongoOp.GREATER_OR_EQUAL.getCompareOp())
+    .put(SqlStdOperatorTable.LESS_THAN, MongoOp.LESS.getCompareOp())
+    .put(SqlStdOperatorTable.LESS_THAN_OR_EQUAL, MongoOp.LESS_OR_EQUAL.getCompareOp())
+    .build();
+
+
+  protected RexToMongoTranslator(JavaTypeFactory typeFactory,
+      List<String> inFields) {
+    super(true);
+    this.typeFactory = typeFactory;
+    this.inFields = inFields;
+  }
+
+  @Override
+  public BsonValue visitLiteral(RexLiteral literal) {
+    if (literal.getValue() == null) {
+      return BsonNull.VALUE;
+    }
+    return new BsonDocument("$literal", new BsonString(
+        RexToLixTranslator.translateLiteral(literal, literal.getType(),
+            typeFactory, RexImpTable.NullAs.NOT_POSSIBLE).toString()));
+  }
+
+  @Override
+  public BsonValue visitInputRef(RexInputRef inputRef) {
+    return new BsonString("$" + inFields.get(inputRef.getIndex()));
+  }
+
+  @Override
+  public BsonValue visitCall(RexCall call) {
+    String name = isItem(call);
+    if (name != null) {
+      return new BsonString("'$" + name + "'");
+    }
+    List<BsonValue> strings = call.operands.stream()
+        .map(operand -> operand.accept(this))
+        .collect(Collectors.toList());
+
+    if (call.getKind() == SqlKind.CAST) {
+      return strings.get(0);
+    }
+    String stdOperator = MONGO_OPERATORS.get(call.getOperator());
+    if (stdOperator != null) {
+      return new BsonDocument(stdOperator, new BsonArray(strings));
+    }
+    if (call.getOperator() == SqlStdOperatorTable.ITEM) {
+      RexNode op1 = call.operands.get(1);
+      if (op1 instanceof RexLiteral) {
+        if (op1.getType().getSqlTypeName() == SqlTypeName.INTEGER) {
+          return new BsonDocument("$arrayElemAt", new BsonArray(
+              Arrays.asList(strings.get(0), new BsonInt32(((RexLiteral) op1).getValueAs(Integer.class)))));
+        } else if (op1.getType().getSqlTypeName() == SqlTypeName.CHAR) {
+          return new BsonString(strings.get(0).asString().getValue() + "." + ((RexLiteral) op1).getValueAs(String.class));
+        }
+      }
+    }
+    if (call.getOperator() == SqlStdOperatorTable.CASE) {
```

Comment:
It doesn't look like there is a test for this multi-leg case translation, can we add one?

Reply:
Thanks, added